framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -869,7 +869,11 @@ public AnnotationFormatter getAnnotationFormatter() {
      *         empty set if no qualifiers are supported
      */
     public final Set<Class<? extends Annotation>> getSupportedTypeQualifiers() {
-        return supportedQuals;
+        if (this.supportedQuals.isEmpty()) {
+            supportedQuals.addAll(createSupportedTypeQualifiers());
+            checkSupportedQuals();
+        }
+        return Collections.unmodifiableSet(supportedQuals);
```

Comment:
`createSupportedTypeQualifiers` is already specified to return an immutable set.
Now that this field has to contain a mutable set, this is redundant.
Can you remove all the `unmodifiableSet` calls and specifications from `createSupportedTypeQualifiers` and all methods it calls?